src/main/java/com/hazelcast/aws/AwsDiscoveryStrategy.java

```diff
@@ -83,23 +85,45 @@ public AwsDiscoveryStrategy(Map<String, Comparable> properties) {
         this.awsClient = client;
     }
 
-    private AwsConfig getAwsConfig()
+    /**
+     * For test purposes only.
+     */
+    AwsDiscoveryStrategy(Map<String, Comparable> properties, AwsConfig awsConfig, AWSClient client) {
+        super(LOGGER, properties);
+        this.awsConfig = awsConfig;
+        this.awsClient = client;
+    }
+
+    AwsConfig getAwsConfig()
             throws IllegalArgumentException {
+        Integer connectionRetries = getOrDefault(CONNECTION_RETRIES.getDefinition(), DEFAULT_CONNECTION_RETRIES);
+        Integer connectionTimeoutSeconds = getOrDefault(CONNECTION_TIMEOUT_SECONDS.getDefinition(),
+                DEFAULT_CONNECTION_TIMEOUT_SECONDS);
+        String region = getOrDefault(REGION.getDefinition(), null);
+        //to prevent unnecessary metadata call when region is set
+        if (region == null) {
+            region = getCurrentRegion(connectionTimeoutSeconds, connectionRetries);
```

Comment:
How about adding some logging here, like "Region not set, using the current region eu-central1". Or something similar?

Reply:
Added 👍